core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -208,7 +208,7 @@ memory_instantiate(WASMModuleInstance *module_inst, uint32 num_bytes_per_page,
         num_bytes_per_page += heap_size;
         if (num_bytes_per_page < heap_size) {
             set_error_buf(error_buf, error_buf_size,
-                          "memory size must be at most 65536 pages (4GiB)");
+                          "failed to insert app heap into linear memory");
```

Comment:
shall we append something like "maybe use --heap_size=0 and try again"?

Reply:
OK, let's add more prompt.